workshop/content/basics/operations/troubleshooting-provisioning-errors/_index.md

```diff
@@ -0,0 +1,116 @@
+
+### Start Lab
+You will use an example CloudFormation template, that contains an incorrect resource configuration, to create a new stack. You will choose to *preserve successfully provisioned resources* as part of *stack failure options* to preserve the state of an example `ExampleDeadLetterQueue` [Amazon SQS](https://aws.amazon.com/sqs/) queue resource, that will be successfully provisioned as part of the stack creation operation.
+
+The creation operation for your stack will fail because another SQS queue described in your template, `ExampleSourceQueue`, has a configuration error. You will troubleshoot and fix the error in the template, and then you will choose to resume the stack creation operation with the template you updated.
+
+To get started, follow steps shown next:
+1. Change directory to the `code/workspace/troubleshooting-provisioning-errors` directory.
+2. Open the `example_sqs_queues.yaml` CloudFormation template in your favorite text editor.
+3. Familiarize with the configuration for sample SQS queues in the template; your intents, in this example, are to:
+    1. create a *source* queue and a [*dead-letter* queue](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-dead-letter-queues.html) (DLQ): you choose to have your `ExampleSourceQueue` target your `ExampleDeadLetterQueue` for messages that cannot be successfully processed.  In the template, you reference the [Amazon Resource Name](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) (ARN) for  `ExampleDeadLetterQueue` in the [redrive policy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sqs-queues.html#aws-sqs-queue-redrive) for `ExampleSourceQueue`: your DLQ will then be created first, so that your *source* queue can then reference its ARN;
+    2. describe both queues as [First-In-First-Out](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sqs-queues.html#aws-sqs-queue-fifoqueue) (FIFO) queues in this example. When you describe a *source* queue and a *dead-letter* [queue](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sqs-queues.html#aws-sqs-queue-redrive), both queues must be of the same type: either *standard* or *FIFO* (the latter is the example for this lab). You note that the example template contains different configuration values for the `FifoQueue` property of your 2 queues, for which you instead expect to have a value of `true` for both. Moreover, when you describe a FIFO queue, its name must contain a `.fifo` suffix: whilst `ExampleSourceQueue` contains the `.fifo` suffix in its `QueueName`, it is configured as a standard queue (i.e., with `FifoQueue` set to `false`), thus resulting in an error.
```

Comment:
```suggestion
    2. describe both queues as [First-In-First-Out](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sqs-queues.html#aws-sqs-queue-fifoqueue) (FIFO) queues in this example. When you describe a *source* queue and a *dead-letter* [queue](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sqs-queues.html#aws-sqs-queue-redrive), both queues must be of the same type: either *standard* or *FIFO* (the latter is the example for this lab). Note, that the example template contains different configuration for the two `FifoQueue` property values, instead of expected value of `true` for both. Moreover, when you describe a FIFO queue, its name must contain a `.fifo` suffix: whilst `ExampleSourceQueue` contains the `.fifo` suffix in its `QueueName`, it is configured as a standard queue (i.e., with `FifoQueue` set to `false`), thus resulting in an error.
```